libs/libc/obstack/lib_obstack_room.c

```diff
@@ -0,0 +1,41 @@
+/****************************************************************************
+ * libs/libc/obstack/lib_obstack_room.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <obstack.h>
+
+#include <assert.h>
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+size_t obstack_room (FAR struct obstack *h)
+{
+  DEBUGASSERT(h);
+
+  if (h->chunk)
+      return h->chunk->limit - h->next_free;
+
+  return 0;
```

Comment:
```suggestion
  return h->chunk != NULL ? h->chunk->limit - h->next_free : 0;
```
Or maybe even a macro

Reply:
I am not convinced about macros that result in multiple expansions, and I am unaware of a way to implement it in C99 without it as a macro. We can either accept here triple expansion or use a function. In the end, it is up to you.